### PR TITLE
🕶 Dark mode for reports

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?
This is dark mode for the Reports screen + the selection screen (it was necessary for the workspace selection, I hope it won't cause merge conflicts).

### Relationships
Ref #5969

## Why do we want this?
We want dark mode in the next release.

## How is it done?
Using `ColorAssets`. I tried to set most of the colors in the interface builder and change stuff in code only when it was necessary (e.g., backgrounds of table views).

The view page should look good in the Dark mode and the "new light look". There were no designs and I hope it looks good 😄.

### Why not another way?
There is no other way.

### Side effects
Some people might not like it (or some details of it).

## Review hints
Try it in the simulator, both the light and dark modes. Here are some screenshots:

| Dark | Light |
|:----:|:----:|
| image | image |
| image |  image| 
|  image | image |
| image | image |
| image |  image |

I think we should merge this ASAP to avoid conflicts and tune the colors with a designer or with the product manager(s) and tune it later.

## :squid: Permissions
@Odrakir

### CODE REVIEW

**Comment by simonrozsival** [comment]
BTW the iPad version is kinda broken and sad :D 
image
image

image
image


... but the white background stripes on the sides are on the timer page as well, so I guess we can fix it later (@dshr could help with that - where do I set that background color? I'm not sure where to look for it and it might be very easy, so please show me :)

**Comment by simonrozsival** [comment]
(also the borders on the iPad are kinda weird - how to get rid of those?)

File: Toggl.iOS/Extensions/ColorAssets.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@ public class ColorAssets
     {
         public static readonly UIColor Background = UIColor.FromName("Background");
         public static readonly UIColor Text = UIColor.FromName("Text");
+        public static readonly UIColor InverseText = UIColor.FromName("InverseText");
```
Comment (Odrakir) [inline Toggl.iOS/Extensions/ColorAssets.cs]: This one is basically the same as `Background`, right? I'm not saying we shouldn't have it, but if we go down that road we might end up with a bunch of colors that are the same but with different names. What's your thinking? Should we instead have just one with another name?

File: Toggl.iOS/Cells/Common/SelectorCell.cs [hunk 2]
```diff
@@ -34,6 +35,7 @@ protected override void UpdateView()
         {
             NameLabel.Text = Item;
             CheckMark.Hidden = !OptionSelected;
+            BackgroundColor = ColorAssets.CellBackground;
```
Comment (Odrakir) [inline Toggl.iOS/Cells/Common/SelectorCell.cs]: In the case of cells I'm also removing the separator that's created in IB and adding it in code with the `InsertSeparator` extension method. This way they'll be 1 pixel thick, and not 1 point thick.

File: Toggl.iOS/Views/Reports/ReportsCalendarViewCell.cs [hunk 3]
```diff
@@ -42,28 +42,27 @@ private void prepareViews()
...
-
-            if (Item.IsSelected(dateRange))
-            {
-                Text.TextColor = selectedColor;
-            }
-            else
-            {
-                Text.TextColor = Item.IsInCurrentMonth ? thisMonthColor : otherMonthColor;
-            }
+            Text.TextColor = (Item.IsToday && !Item.IsSelected(dateRange))
```
Comment (Odrakir) [inline Toggl.iOS/Views/Reports/ReportsCalendarViewCell.cs]: I am dizzy reading this...

File: Toggl.iOS/Extensions/ColorAssets.cs (hunk 1 above)
Comment (simonrozsival) [reply Toggl.iOS/Extensions/ColorAssets.cs]: I'd like to keep it. I know it's the same and I thought about it for a while, but I didn't want to use a color which is used for backgrounds as a text color. If you don't think it's a good reason, we can get rid of it.

**Pull request merged**